### DISCUSSION TITLE
A minor detail about positional embedding

### DISCUSSION THREAD

**Discussion by Yijia-Xiao** [facebookresearch/esm#133] [Q&A] [answered]
Hi, I found an interesting phenomenon in the process of using your MSA Transformer model.
In the [LearnedPositionalEmbedding](https://github.com/facebookresearch/esm/blob/main/esm/modules.py#L246) class, you added `padding_idx` to `positions` (kind of like offset). I wonder why padding_idx is added and is this addition necessary? Thank you!

**Comment by tomsercu** [comment, marked as answer]
This is borrowed directly from [fairseq](https://github.com/pytorch/fairseq/blob/main/fairseq/modules/learned_positional_embedding.py).
From a quick skim the crux seems to be to have a specific index in the learned embedding (ie `self.padding_idx=1`) which learns the embedding for padding. The actual positional embeddings (in our case 0-1023) are shifted to avoid re-using it (+2), so the total number of learned embedding vectors is 1024+2

**Comment by Yijia-Xiao** [comment]
Got it! Thank you for your explanation :)
